Gem/Code/Source/Components/AWSMetricsSubmissionComponent.cpp

```diff
@@ -0,0 +1,129 @@
+                ->Version(1);
+
+            if (const auto editContext = serializationContext->GetEditContext())
+            {
+                editContext->Class<AWSMetricsSubmissionComponent>("AWSMetricsSubmissionComponent",
+                    "Handles metrics submission via the AWSMetrics Gem")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::Category, "MultiplayerSample")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("Game"))
+                    ->DataElement(nullptr, &AWSMetricsSubmissionComponent::m_AWSMetricsSubmissionIntervalInSecHint, "AWS Metrics Submission Interval Hint in Seconds",
+                        "Hint for the AWS metrics submission interval. "
+                        "This interval will apply to the metrics that need to be submitted regulary like player connection count.")
+                    ;
+            }
+        }
+    }
+
+    void AWSMetricsSubmissionComponent::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided)
+    {
+        provided.push_back(AZ_CRC_CE("AWSMetricsSubmissionComponent"));
+    }
+
+    void AWSMetricsSubmissionComponent::GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible)
+    {
+        incompatible.push_back(AZ_CRC_CE("AWSMetricsSubmissionComponent"));
+    }
+
+    void AWSMetricsSubmissionComponent::Activate()
+    {
+        if (!sv_submit_aws_metrics)
+        {
+            return;
+        }
+
+        RegisterPlayerConnectionMetrics();
+
+        AZ::TickBus::Handler::BusConnect();
+    }
+
+    void AWSMetricsSubmissionComponent::Deactivate()
+    {
+        if (!sv_submit_aws_metrics)
+        {
+            return;
+        }
+
+        AZ::TickBus::Handler::BusDisconnect();
+
+        AWSMetrics::AWSMetricsRequestBus::Broadcast(&AWSMetrics::AWSMetricsRequests::FlushMetrics);
+    }
+
+    void AWSMetricsSubmissionComponent::RegisterPlayerConnectionMetrics()
+    {
+        // Add a handler for the ConnectionAcquired event to submit the client_join metrics
+        m_connectHandler = ConnectionAcquiredEvent::Handler([](MultiplayerAgentDatum)
+            {
+                AZStd::vector<AWSMetrics::MetricsAttribute> clientJointMetricsAttributes;
+                clientJointMetricsAttributes.emplace_back(AWSMetrics::MetricsAttribute("event_name", "client_join"));
+
+                AWSMetrics::AWSMetricsRequestBus::Broadcast(&AWSMetrics::AWSMetricsRequests::SubmitMetrics, clientJointMetricsAttributes, 0, AZStd::string("MultiplayerSample"), true);
+            });
+        AZ::Interface<IMultiplayer>::Get()->AddConnectionAcquiredHandler(m_connectHandler);
+
+        // Add a handler for the EndpointDisonnected event to submit the client_leave metrics
+        m_disconnectHandler = EndpointDisonnectedEvent::Handler([](MultiplayerAgentType)

```

Comment:
this requires changes made in https://github.com/o3de/o3de/pull/10401 , yes?

Reply:
Correct. this requires https://github.com/o3de/o3de/pull/10401https://github.com/o3de/o3de/pull/10401https://github.com/o3de/o3de/pull/10401 to be submitted first.